protecode/util.py

```diff
@@ -114,13 +115,22 @@ def group_resources(components):
             for resource_name, resources in resource_groups.items():
                 yield resources
 
+        def _has_required_scan_label(resource):
+            for label in product.v2.scan_labels(resource.access):
```

Comment:
I think it might make sense to add a util-function that will already do the filtering (similar to `scan_labels`). in fact, there is already such a method :-)))

--> https://github.com/gardener/component-spec/blob/master/bindings-python/gci/componentmodel.py#L105